### PR TITLE
Chromium60.0.3112.50

### PR DESCRIPTION
**Pull request by hferreiro**
Update to Chromium 60.0.3112.50

### CODE REVIEW

File: atom/browser/api/atom_api_app.cc [hunk 1]
```diff
@@ -47,7 +47,6 @@
 #include "content/public/browser/browser_thread.h"
 #include "content/public/browser/client_certificate_delegate.h"
 #include "content/public/browser/gpu_data_manager.h"
-#include "content/public/browser/memory_pressure_controller.h"
```
Comment (bridiver) [inline atom/browser/api/atom_api_app.cc]: why this change? If `MemoryPressureListener` moved we still need an include for it

File: atom/browser/api/atom_api_menu_views.cc [hunk 2]
```diff
@@ -43,9 +43,7 @@ void MenuViews::PopupAt(
     location = gfx::Point(origin.x() + x, origin.y() + y);
   }
 
-  int flags = MenuRunner::CONTEXT_MENU |
-              MenuRunner::HAS_MNEMONICS |
-              MenuRunner::ASYNC;
+  int flags = MenuRunner::CONTEXT_MENU | MenuRunner::HAS_MNEMONICS;
```
Comment (bridiver) [inline atom/browser/api/atom_api_menu_views.cc]: @bsclifton is this change ok?

File: atom/browser/api/atom_api_web_contents.cc [hunk 3]
```diff
@@ -2037,12 +2035,8 @@ bool WebContents::ExecuteScriptInTab(mate::Arguments* args) {
 
 bool WebContents::SendIPCSharedMemory(const base::string16& channel,
                                       base::SharedMemory* shared_memory) {
-  auto process_handle = web_contents()->GetRenderProcessHost()->GetHandle();
-  if (!process_handle)
-    return false;
-
-  base::SharedMemoryHandle memory_handle;
-  if (!shared_memory->ShareToProcess(process_handle, &memory_handle))
+  base::SharedMemoryHandle memory_handle = shared_memory->handle().Duplicate();
```
Comment (bridiver) [inline atom/browser/api/atom_api_web_contents.cc]: this change looks suspicious to me. We don't have to share the handle anymore?

File: atom/browser/api/atom_api_menu_views.cc (hunk 2 above)
Comment (darkdh) [inline atom/browser/api/atom_api_menu_views.cc]: @bridiver , yeah per
https://github.com/brave/muon/pull/161#discussion_r120406129

File: brave/browser/password_manager/brave_password_manager_client.cc [hunk 4]
```diff
@@ -339,6 +339,20 @@ BravePasswordManagerClient::GetPasswordStore() const {
              profile_, ServiceAccessType::EXPLICIT_ACCESS).get();
 }
 
+#if defined(SAFE_BROWSING_DB_LOCAL)
```
Comment (bridiver) [inline brave/browser/password_manager/brave_password_manager_client.cc]: I thought this change was already in master? @darkdh ?

File: patches/master_patch.patch [hunk 5]
```diff
@@ -907,17 +888,10 @@ index 7e05c4e04f8e61b418d373a73bd8901347df4c76..e7e6d9fa5e0d54c7755fe3a38eca6f70
...
 +++ b/chrome/utility/importer/firefox_importer.cc
-@@ -140,14 +140,12 @@ void FirefoxImporter::StartImport(const importer::SourceProfile& source_profile,
-   if ((items & importer::PASSWORDS) && !cancelled()) {
-     bridge_->NotifyItemStarted(importer::PASSWORDS);
-     ImportPasswords();
--    bridge_->NotifyItemEnded(importer::PASSWORDS);
-   }
-   if ((items & importer::AUTOFILL_FORM_DATA) && !cancelled()) {
-     bridge_->NotifyItemStarted(importer::AUTOFILL_FORM_DATA);
+@@ -147,7 +147,6 @@ void FirefoxImporter::StartImport(const importer::SourceProfile& source_profile,
```
Comment (bridiver) [inline patches/master_patch.patch]: cc @darkdh to check

File: brave/browser/password_manager/brave_password_manager_client.cc (hunk 4 above)
Comment (darkdh) [inline brave/browser/password_manager/brave_password_manager_client.cc]: It was removed in https://github.com/brave/muon/commit/8316ae196eee53a7774945458ea86446ced19556#diff-11a9200d30980e9b495a6efb898420af

File: patches/master_patch.patch [hunk 6]
```diff
@@ -1548,17 +1527,7 @@ index 85ece52a001652a3414a47189b2711f215a66290..960abaeab72bbf45b8b460fbbaf1cdb4
...
-   // Set the default color to be white. This is the wrong thing to do, but many
-   // UI components expect this view to be opaque.
-+#ifdef MUON_CHROMIUM_BUILD
-   [background_layer_ setBackgroundColor:CGColorGetConstantColor(kCGColorWhite)];
-+#endif
-   [cocoa_view_ setLayer:background_layer_];
-   [cocoa_view_ setWantsLayer:YES];
- 
-@@ -1856,7 +1864,14 @@ - (BOOL)acceptsMouseEventsWhenInactive {
+@@ -1863,7 +1869,14 @@ void RenderWidgetHostViewMac::OnDisplayMetricsChanged(
```
Comment (bridiver) [inline patches/master_patch.patch]: is this change right? Sometimes these move around and can be hard to track down

File: patches/master_patch.patch [hunk 7]
```diff
@@ -1600,42 +1569,23 @@ index 85ece52a001652a3414a47189b2711f215a66290..960abaeab72bbf45b8b460fbbaf1cdb4
...
-+#ifdef MUON_CHROMIUM_BUILD
-         baselinePoint.x += view_box.origin().x() - root_box.origin().x();
-         baselinePoint.y +=
-             root_box.bottom_left().y() - view_box.bottom_left().y();
-+#endif  // MUON_CHROMIUM_BUILD
-         [self showLookUpDictionaryOverlayInternal:string
-                                     baselinePoint:baselinePoint
-                                        targetView:targetView];
-@@ -2835,6 +2858,11 @@ - (RenderWidgetHostViewMac*)renderWidgetHostViewMac {
+@@ -2838,6 +2855,11 @@ void RenderWidgetHostViewMac::OnDisplayMetricsChanged(
```
Comment (bridiver) [inline patches/master_patch.patch]: same as above, this change looks suspicious

File: patches/master_patch.patch [hunk 8]
```diff
@@ -2021,34 +1928,55 @@ index 5374bf1652af381a1efc0cdfc96b56b883b716a0..b157bc09a570ca3dd492ffcfb03417e3
...
 diff --git a/third_party/WebKit/Source/core/html/HTMLMediaElement.cpp b/third_party/WebKit/Source/core/html/HTMLMediaElement.cpp
-index 2fc0977882d944317b26a89c095cde076383e308..223eab775d985f06d1775e5141bd4aae6eb5f0d2 100644
+index 4586eb16224502b5ac6f3bfe50d7d5072b708562..ac8b4793be3b631023a16396a35ed7983e0916ba 100644
 --- a/third_party/WebKit/Source/core/html/HTMLMediaElement.cpp
 +++ b/third_party/WebKit/Source/core/html/HTMLMediaElement.cpp
-@@ -2626,7 +2626,7 @@ void HTMLMediaElement::setMuted(bool muted) {
-     }
-   }
+@@ -2495,7 +2495,7 @@ void HTMLMediaElement::setMuted(bool muted) {
+   if (!muted_ && !autoplay_policy_->RequestAutoplayUnmute())
```
Comment (bridiver) [inline patches/master_patch.patch]: cc @darkdh to check

File: vendor/brightray/browser/network_delegate.cc [hunk 9]
```diff
@@ -144,4 +144,24 @@ bool NetworkDelegate::OnCancelURLRequestWithPolicyViolatingReferrerHeader(
   return false;
 }
 
+bool NetworkDelegate::OnCanQueueReportingReport(
+    const url::Origin& origin) const {
+  return true;
```
Comment (bridiver) [inline vendor/brightray/browser/network_delegate.cc]: these should definitely all be `true`?

File: DEPS [hunk 10]
```diff
@@ -1,7 +1,7 @@
 use_relative_paths = True
 
 deps = {
-  "vendor/node": "https://github.com/brave/node.git@f51b9ab8ff446ca7b13be0de1bc12b854b23938d",
+  "vendor/node": "https://github.com/brave/node.git@88df98b8cbee4a6ee3ba30baaf27f16a81b2360a",
```
Comment (bridiver) [inline DEPS]: node changes? Is there a PR?

File: patches/master_patch.patch (hunk 5 above)
Comment (darkdh) [inline patches/master_patch.patch]: the line is removed intentionally because of https://github.com/brave/muon/blob/master/brave/utility/importer/firefox_importer.cc#L45
so we have to keep that patch

File: atom/browser/api/atom_api_app.cc (hunk 1 above)
Comment (hferreiro) [inline atom/browser/api/atom_api_app.cc]: memory_pressure_controller.h was deleted in https://chromium.googlesource.com/chromium/src/+/37c4e81eb4dfa713a06164dbce721710b92b2ca9
I'll explicitly include "base/memory/memory_pressure_listener.h", I thought this was checked by cpplint

File: atom/browser/api/atom_api_web_contents.cc (hunk 3 above)
Comment (hferreiro) [inline atom/browser/api/atom_api_web_contents.cc]: Since https://chromium.googlesource.com/chromium/src/+/37e7d8301a078bd1b6b3ef7cffdb6c6ac3277edd, ShareToProcess is not needed anymore. Apparently, just duplicating the handle is enough.

File: patches/master_patch.patch (hunk 5 above)
Comment (hferreiro) [inline patches/master_patch.patch]: I guess this was automatically merged by git when rebasing on top of muon. Would you mind pinging me when making changes to files in patches/?

File: patches/master_patch.patch (hunk 6 above)
Comment (hferreiro) [inline patches/master_patch.patch]: Setting the background color to white was removed here: https://chromium.googlesource.com/chromium/src/+/e1979f60524da9dab9d8cc19a23910f8d5dcf180

File: patches/master_patch.patch (hunk 7 above)
Comment (bridiver) [inline patches/master_patch.patch]: although to be honest I'd like to get rid of these patches, just want to make sure we don't break anything

File: patches/master_patch.patch (hunk 5 above)
Comment (darkdh) [inline patches/master_patch.patch]: sure thing

File: patches/master_patch.patch (hunk 8 above)
Comment (darkdh) [inline patches/master_patch.patch]: @hferreiro , why will we need this?
It is already in https://chromium.googlesource.com/chromium/src.git/+/12d76adf4bb340d8e430a18c3a664940157800f6/third_party/WebKit/Source/core/html/HTMLMediaElement.cpp#2458

File: patches/master_patch.patch (hunk 7 above)
Comment (hferreiro) [inline patches/master_patch.patch]: I can't remember why I thought I could remove that patch. An update to that code was introduced here: https://chromium.googlesource.com/chromium/src/+/67a361af4485aee793aa2ad8dacddd249d19fc1c%5E%21/#F3.
I don't have any context for the patch motivation so it is hard to understand its purpose.

File: patches/master_patch.patch (hunk 8 above)
Comment (hferreiro) [inline patches/master_patch.patch]: That is only an update to the patch context, the patch only updates a comment. The original master_patch.patch contained this:
```
--- a/third_party/WebKit/Source/core/html/HTMLMediaElement.cpp
+++ b/third_party/WebKit/Source/core/html/HTMLMediaElement.cpp
@@ -2626,7 +2626,7 @@ void HTMLMediaElement::setMuted(bool muted) {
     }
   }
 
-  // This is called after the volumechange event to make sure isAutoplayingMuted
+  // This is called after the volumechange event to make sure IsAutoplayingMuted
```
which, as you said, now looks like this: https://chromium.googlesource.com/chromium/src.git/+/12d76adf4bb340d8e430a18c3a664940157800f6/third_party/WebKit/Source/core/html/HTMLMediaElement.cpp#2461.

It is quite confusing to review changes to a patch, we should consider moving all of it to proper commits.

Comment (darkdh) [inline patches/master_patch.patch]: I see now. sorry my bad and I think we can remove the comment patch.

File: vendor/brightray/browser/network_delegate.cc (hunk 9 above)
Comment (hferreiro) [inline vendor/brightray/browser/network_delegate.cc]: I just looked at how the other functions were implemented and, similarly, as how [chrome_network_delegate.cc](https://chromium.googlesource.com/chromium/src/+/master/chrome/browser/net/chrome_network_delegate.cc) delegates to cookie_settings_->IsCookieAccessAllowed().

File: DEPS (hunk 10 above)
Comment (hferreiro) [inline DEPS]: Not yet. What branch are you tracking?

File: vendor/brightray/browser/network_delegate.cc (hunk 9 above)
Comment (bridiver) [inline vendor/brightray/browser/network_delegate.cc]: I think I'm just going to dump this class and replace with network_delegate_impl

File: patches/master_patch.patch (hunk 7 above)
Comment (bridiver) [inline patches/master_patch.patch]: https://github.com/electron/libchromiumcontent/commit/7e74297da3fa534129e2b716bce4112940135aec

Comment (bridiver) [inline patches/master_patch.patch]: I'll test it and see if there are any issues without it

Comment (bridiver) [inline patches/master_patch.patch]: first click seems to work fine

Comment (bridiver) [inline patches/master_patch.patch]: mouse cursor seems fine https://github.com/electron/electron/issues/995

Comment (bridiver) [inline patches/master_patch.patch]: don't care about window transparency https://github.com/electron/electron/pull/949

Comment (bridiver) [inline patches/master_patch.patch]: @hferreiro moving to a commit is somewhat difficult. We maintain a pseudo-fork for android that doesn't have all the history and that's a possibility, but patches are a last resort and we're tying to remove most of them so not sure if it's really worth the time right now

Comment (bridiver) [inline patches/master_patch.patch]: menu shortcuts also appear to be ok https://github.com/electron/electron/issues/170

Comment (bridiver) [inline patches/master_patch.patch]: actually disable auto hide is still in there

**Comment by bridiver** [comment]
this is already merged into master via another branch so I'm going to close. Thanks!
